spring-cloud-app-broker-deployer-cloudfoundry/src/main/java/org/springframework/cloud/appbroker/deployer/cloudfoundry/CloudFoundryAppDeployer.java

```diff
@@ -941,6 +941,7 @@ private Path getApplication(Resource resource) {
 				.serviceName(request.getName())
 				.planName(request.getPlan())
 				.parameters(request.getParameters())
+				.completionTimeout(Duration.ofSeconds(this.defaultDeploymentProperties.getApiPollingTimeout()))
```

Comment:
This code will just set the default to the time specified in, which is great but not what was documented.
If we want a property to be passed through we need to do something like https://github.com/spring-cloud/spring-cloud-app-broker/blob/master/spring-cloud-app-broker-deployer-cloudfoundry/src/main/java/org/springframework/cloud/appbroker/deployer/cloudfoundry/CloudFoundryAppDeployer.java#L792-L796 where we check if we have a value that was passed to the `Deployer` layer, otherwise, we use the provided default.

Reply:
Thanks for catching this. I'm preparing a fix to indeed fetch the configured from properties.

> Also, note that all the properties passed to CloudFoundryDeploymentProperties were always application-related this will be the first and only one related to services.

Please let me know if this would be ok to leave this property generically named in CloudFoundryDeploymentProperties, or what alternative would be recommended instead.

Note that Javadoc seems to assume brokered services could be configured with this property https://github.com/spring-cloud/spring-cloud-app-broker/blob/171554905e2a6eb39d4082be42d131fd33a0a4e7/spring-cloud-app-broker-deployer-cloudfoundry/src/main/java/org/springframework/cloud/appbroker/deployer/cloudfoundry/CloudFoundryDeploymentProperties.java#L28-L29

Reply:
Go with the name you added. Thanks for explaining!

Reply:
Thanks.

The fix to fetch the timeout from configured properties is in  8455458
I'm therefore resolving the conversation, please reopen if there is something missing.